spec/jobs/dashboard/updater_job_spec.rb

```diff
@@ -10,8 +10,10 @@ module Dashboard
 
         context "and job is not in the suspended list" do
           it "instantiates widget with the specified parameter" do
-            expect(Dashboard::Widget).to receive(:new).with("MyWidget").and_return(instance_double(Dashboard::SingleObject::Feedback, run: nil))
+            widget = instance_double(Dashboard::SingleObject::Feedback, run: nil)
+            allow(Dashboard::Widget).to receive(:new).with("MyWidget").and_return(widget)
             described_class.perform_now("MyWidget")
+            expect(widget).not_to receive(:run)
```

Comment:
don't you need to use `have_received` after a call, or put expectation`to receive` before the call

```suggestion
            expect(widget).not_to have_received(:run)
```

Reply:
Ran the update and I made a mistake when I extracted it... Now changed to 
```expect(widget).to have_received(:run)```
The not_to was a mistake but as it wasn't exercised properly it wasn't failing the test